File: Taller RabbitMQ/consumer_php/consumer.php
<?php

require_once __DIR__ . '/vendor/autoload.php';

use PhpAmqpLib\Connection\AMQPStreamConnection;

// Configuración de la conexión
$host = 'rabbitmq';
$port = 5672;
$user = 'guest';
$pass = 'guest';
$queue = 'cola1';

// Crear conexión
$connection = new AMQPStreamConnection($host, $port, $user, $pass);
$channel = $connection->channel();

// Declarar la cola
$channel->queue_declare($queue, false, false, false, false);

echo "Esperando mensajes desde PHP. Para salir, presiona CTRL+C\n";

// Callback para procesar los mensajes
$callback = function ($msg) {
    echo "Consumer PHP: Mensaje recibido -> '" . $msg->body . "'\n";
};

// Consumir mensajes de la cola
$channel->basic_consume($queue, '', false, true, false, false, $callback);

// Mantener el script en ejecución
while ($channel->is_consuming()) {
    $channel->wait();
}

// Cerrar conexión (esto no se ejecutará a menos que se detenga manualmente)
$channel->close();
$connection->close();

File: Taller RabbitMQ/consumer_python/consumer.py
import pika

# Conexión a RabbitMQ
connection = pika.BlockingConnection(pika.ConnectionParameters('rabbitmq'))
channel = connection.channel()

# Declarar la cola
queue = 'hello'
channel.queue_declare(queue=queue, durable=False)

print("Esperando Mensajes desde Python. Para salir, presione CTRL+C")

# Callback para procesar los mensajes recibidos
def callback(ch, method, properties, body):
    print(f"Consumer Python: {body.decode()}")

# Consumir mensajes de la cola
channel.basic_consume(queue=queue,
                      on_message_callback=callback,
                      auto_ack=True)

try:
    channel.start_consuming()
except KeyboardInterrupt:
    print("Consumer Python detenido.")
finally:
    connection.close()
